### PR TITLE
Additional Routines

### PR DESCRIPTION
**Pull request by salomon73**
I added a few Matlab routines essentially related to straight field line coordinates, as well as boozer coordinates. Others are related to the magnetic shear and enable to extract it from spec output, as well as scan this parameter over some plasma boundary Fourier modes.

Finally, some python routines were added, enabling to achieve Boozer coordinates transformation, and plot modB in this coordinates system, from SPEC output (.h5 file)

### CODE REVIEW

**Comment by salomon73** [comment]
Only Utilities directory has been changed

**Comment by abaillod** [comment]
Hi Salomon,

I did not see you started a PR! I will add some reviewers so we can move forward with this. Thanks for the time you took to prepare these new tools.

I think we can safely merge this, it is only additional routines, and there are no modifications of other parts of the code. We should probably remove the file Utilities/matlabtools/scan_shear_auto_optim.m though, this is specific to the study you did.

File: Utilities/BoozerXForms/init_from_spec.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    for ii in range(0,mnmax):
+        m = xm[ii]
+        n = int(xn[ii] / nfp)
+        Bmnc[ii,jj] = Bmn_[m,n+ntor]
+        
+    Bmns = np.array([])
+
+
+    ##########################################################
+    # initialisation de l'instance b de la classe booz_xform #
```
Comment (jonathanschilling) [inline Utilities/BoozerXForms/init_from_spec.py:122]: Comment should be in English, please :-)

Comment (salomon73) [reply Utilities/BoozerXForms/init_from_spec.py:122]: I'll do it ! My bad :)

**Comment by salomon73** [comment]
Hi all, 
anything left that I should do in order for the pull request to be merged ? 
Best, 
Salomon

**Comment by abaillod** [comment]
I think you don't have the access to press the green button. I will do it for you!

**Pull request merged**